plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SqlStandardAccessControl.java

```diff
@@ -583,9 +582,16 @@ public void checkCanExecuteTableProcedure(ConnectorSecurityContext context, Sche
     @Override
     public void checkCanExecuteFunction(ConnectorSecurityContext context, FunctionKind functionKind, SchemaRoutineName function)
     {
-        if (functionKind == TABLE && !isAdmin(context)) {
-            denyExecuteFunction(function.toString());
+        switch (functionKind) {
+            case SCALAR, AGGREGATE, WINDOW:
+                return;
+            case TABLE:
+                if (isAdmin(context)) {
+                    return;
+                }
+                denyExecuteFunction(function.toString());
```

Comment:
Add 
>  // TODO implement

?

Reply:
`Make checkCanExecuteFunction checks future proof` is meant to be just a code refactor.